MbsAPI/fLmd.c

```diff
@@ -137,12 +137,7 @@ uint32_t fLmdPutOpen(sLmdControl* pLmdControl,
 
     pLmdControl->pMbsFileHeader->iType = LMD__TYPE_FILE_HEADER_101_1;
     pLmdControl->pMbsFileHeader->iEndian = 1;
-    size_t len = strlen(Filename);
-    if (len < sizeof(pLmdControl->cFile)) {
-        strncpy(pLmdControl->cFile, Filename, len);
-    } else {
-        strncpy(pLmdControl->cFile, Filename, sizeof(pLmdControl->cFile) - 1);
-    }
+    snprintf(pLmdControl->cFile, sizeof(pLmdControl->cFile), "%s", Filename);
```

Comment:
Maybe one could add a warning when the length of the source is larger than the destination. Otherwise the filename is truncated without any information. This should have done already in the past.

The same holds for all the other places were this pattern appears.

Reply:
Actually I touched the code in the past to avoid strcpy. Obviously I didn't understood strncpy and did not improve the code but only got rid of the warning.

Reply:
Well, strncpy is already an improvement over strcpy, as it doesn't write past the end of the buffer.

My original intention was to have a minimal patch that doesn't change the intended semantics (truncate without warning).
I have added some printing of warnings in this update. Is it now better?
(I haven't done it for all cases that I touched).

Reply:
I think this is fine. Probably @kresan should also comment on this issue. R3B is meanwhile the main user of the MBSApi.

Reply:
Since @kresan confirmed in the meeting today that R3B doesn't use it any longer please merge it